### PR TITLE
2pc: prep for watchdog

### PR DESCRIPTION
**Pull request by sougou**
In the design, two different functions were named ResolveTransaction, but had different meanings. I've renamed the lower level ResolveTransaction to ConcludeTransaction, which is more fitting.
I've also exported the other ResolveTransaction, and the Atomic flag in the VTGate grpc API.

### CODE REVIEW

File: go/vt/tabletserver/tabletserver.go [hunk 1]
```diff
@@ -786,20 +786,20 @@ func (tsv *TabletServer) SetRollback(ctx context.Context, target *querypb.Target
...
 
-// ResolveTransaction deletes the 2pc transaction metadata
+// ConcludeTransaction deletes the 2pc transaction metadata
 // essentially resolving it.
-func (tsv *TabletServer) ResolveTransaction(ctx context.Context, target *querypb.Target, dtid string) (err error) {
+func (tsv *TabletServer) ConcludeTransaction(ctx context.Context, target *querypb.Target, dtid string) (err error) {
 	return tsv.execRequest(
 		ctx, tsv.QueryTimeout.Get(),
-		"ResolveTransaction", "resolve_transaction", nil,
+		"ConcludeTransaction", "resolve_transaction", nil,
```
Comment (alainjobart) [inline go/vt/tabletserver/tabletserver.go]: resolve -> conclude ?

File: go/cmd/vtgateclienttest/services/success.go [hunk 2]
```diff
@@ -32,11 +32,11 @@ func (c *successClient) Begin(ctx context.Context) (*vtgatepb.Session, error) {
 	}, nil
 }
 
-func (c *successClient) Commit(ctx context.Context, session *vtgatepb.Session) error {
+func (c *successClient) Commit(ctx context.Context, twopc bool, session *vtgatepb.Session) error {
```
Comment (alainjobart) [inline go/cmd/vtgateclienttest/services/success.go]: Shouldn't this file have a ResolveTransaction function too?

And it should be exercised by the various clients, right? Like Java, PHP, python, ... Or is this not wired yet, but is coming soon?

File: go/vt/tabletserver/tabletserver.go (hunk 1 above)
Comment (sougou) [inline go/vt/tabletserver/tabletserver.go]: Done.

File: go/cmd/vtgateclienttest/services/success.go (hunk 2 above)
Comment (alainjobart) [inline go/cmd/vtgateclienttest/services/success.go]: Yes, one code path should return success here. I think you'd do it by having the dtid start with success... and then in that case you return success. Then the client passes the right dtid, and checks it worked. But you need the client side to be implemented for this, so maybe wait until you work on that part...

**Comment by alainjobart** [comment]
LGTM

**Pull request merged**